new-packages/ts-project/__tests__/state.test.ts

```diff
@@ -0,0 +1,695 @@
+            },
+          },
+          state4: {},
+        },
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+  expect(project.extractMachines('index.ts')).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "edges": [],
+          "rootState": {
+            "data": {
+              "description": undefined,
+              "entry": [],
+              "exit": [],
+              "history": undefined,
+              "initial": undefined,
+              "invoke": [],
+              "metaEntries": [],
+              "tags": [],
+              "type": undefined,
+            },
+            "id": "",
+            "states": [
+              {
+                "data": {
+                  "description": undefined,
+                  "entry": [],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": undefined,
+                },
+                "id": "state1",
+                "states": [
+                  {
+                    "data": {
+                      "description": undefined,
+                      "entry": [],
+                      "exit": [],
+                      "history": undefined,
+                      "initial": undefined,
+                      "invoke": [],
+                      "metaEntries": [],
+                      "tags": [],
+                      "type": undefined,
+                    },
+                    "id": "state1.state2",
+                    "states": [
+                      {
+                        "data": {
+                          "description": undefined,
+                          "entry": [],
+                          "exit": [],
+                          "history": undefined,
+                          "initial": undefined,
+                          "invoke": [],
+                          "metaEntries": [],
+                          "tags": [],
+                          "type": undefined,
+                        },
+                        "id": "state1.state2.state3",
+                        "states": [],
+                      },
+                    ],
+                  },
+                ],
+              },
+              {
+                "data": {
+                  "description": undefined,
+                  "entry": [],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": undefined,
+                },
+                "id": "state4",
+                "states": [],
+              },
+            ],
+          },
+          "states": [],
+        },
+        [],
+      ],
+    ]
+  `);
+});
+
+test('Should extract state.initial with string value', async () => {
```

Comment:
nit: the inconsistency with the title titles starting with either uppercase or a lowercase kills me. Personally, I'd prefer sticking to a lowercase.